doc/source/whatsnew/v0.19.0.txt

```diff
@@ -1075,3 +1075,4 @@ Bug Fixes
 - Bug in ``Index`` raises ``KeyError`` displaying incorrect column when column is not in the df and columns contains duplicate values (:issue:`13822`)
 - Bug in ``Period`` and ``PeriodIndex`` creating wrong dates when frequency has combined offset aliases (:issue:`13874`)
 - Bug in ``.to_string()`` when called with an integer ``line_width`` and ``index=False`` raises an UnboundLocalError exception because ``idx`` referenced before assignment.
+- Bug in ``Categorical`` would allow creation when ``object`` dtype was passed in with  categories not containing either all string or all period values
```

Comment:
I don't think it is a bug. Can you move this to API change, and needs more detailed descriptions as it breaks existing user's code.